src/derivative_utils.jl

```diff
@@ -102,27 +246,42 @@ function calc_W!(integrator, cache::OrdinaryDiffEqMutableCache, dtgamma, repeat_
 end
 
 function calc_W!(integrator, cache::OrdinaryDiffEqConstantCache, dtgamma, repeat_step, W_transform=false)
-  @unpack t,uprev,f = integrator
+  @unpack t,uprev,p,f = integrator
   @unpack uf = cache
+  mass_matrix = integrator.f.mass_matrix
   # calculate W
   uf.t = t
   isarray = typeof(uprev) <: AbstractArray
   iscompo = typeof(integrator.alg) <: CompositeAlgorithm
   if !W_transform
-    if isarray
-      J = ForwardDiff.jacobian(uf,uprev)
-      W = I - dtgamma*J
+    if DiffEqBase.has_jac(f)
+      J = f.jac(uprev, p, t)
+      if !isa(f, DiffEqBase.AbstractDiffEqLinearOperator)
+        J = DiffEqArrayOperator(J)
+      end
+      W = WOperator(mass_matrix, dtgamma, J; transform=false)
     else
-      J = ForwardDiff.derivative(uf,uprev)
-      W = 1 - dtgamma*J
+      if isarray
+        J = ForwardDiff.jacobian(uf,uprev)
+      else
+        J = ForwardDiff.derivative(uf,uprev)
+      end
+      W = mass_matrix - dtgamma*J
```

Comment:
out of place isn't using WOperator yet?

Reply:
It's used in the branch where `DiffEqBase.has_jac(f) == true` (https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/443/files/21b1a5b7fae562c8f514cc445002453fa916da6c#diff-35ab888388acf2e7239e17f33e5eefe7R257 and https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/443/files/21b1a5b7fae562c8f514cc445002453fa916da6c#diff-35ab888388acf2e7239e17f33e5eefe7R272).

Reply:
Why not all of the time?

Reply:
I guess it's not necessary all of the time, and if we don't allow a linsolve option on it then it's pointless for now.